client/src/components/CustomFields.js

```diff
@@ -524,59 +588,74 @@ const READONLY_FIELD_COMPONENTS = {
-      const fieldConfig = fieldsConfig[key]
-      const {
-        type,
-        typeError,
-        placeholder,
-        helpText,
-        validations,
-        visibleWhen,
-        objectFields,
-        ...fieldProps
-      } = fieldConfig
-      let extraProps = {}
-      if (type === CUSTOM_FIELD_TYPE.ARRAY_OF_OBJECTS) {
-        extraProps = {
-          fieldConfig,
-          formikProps
+  fieldNamePrefix, // key path in the values object to get to the level of fields given by the fieldsConfig
+  values,
+  vertical
+}) => {
+  return (
+    <>
+      {Object.keys(fieldsConfig).map(key => {
+        const fieldConfig = fieldsConfig[key]
+        const {
+          type,
+          typeError,
+          placeholder,
+          helpText,
+          validations,
+          visibleWhen,
+          objectFields,
+          ...fieldProps
+        } = fieldConfig
+        let extraProps = {}
+        if (type === CUSTOM_FIELD_TYPE.ARRAY_OF_OBJECTS) {
+          extraProps = {
+            fieldConfig
+          }
         }
-      }
-      const FieldComponent = READONLY_FIELD_COMPONENTS[type]
-      return (
-        <FieldComponent
-          key={key}
-          name={`${fieldNamePrefix}.${key}`}
-          {...fieldProps}
-          {...extraProps}
-        />
-      )
-    })}
-  </>
-)
+        const ReadonlyFieldComponent = READONLY_FIELD_COMPONENTS[type]
+        return (
+          <ReadonlyFieldComponent
+            key={key}
+            name={`${fieldNamePrefix}.${key}`}
+            values={values}
+            vertical={vertical}
+            {...fieldProps}
+            {...extraProps}
+          />
+        )
+      })}
+    </>
+  )
+}
 ReadonlyCustomFields.propTypes = {
   fieldsConfig: PropTypes.object,
-  formikProps: PropTypes.object,
-  fieldNamePrefix: PropTypes.string.isRequired
+  fieldNamePrefix: PropTypes.string.isRequired,
+  values: PropTypes.object.isRequired,
+  vertical: PropTypes.bool
 }
 ReadonlyCustomFields.defaultProps = {
-  fieldNamePrefix: DEFAULT_CUSTOM_FIELDS_PREFIX
-}
-
-// customFields should contain the JSON of all the visible custom fields
-export const customFieldsJSONString = values => {
-  if (values.formCustomFields) {
+  fieldNamePrefix: DEFAULT_CUSTOM_FIELDS_PREFIX,
+  vertical: false
+}
+
+// customFields should contain the JSON of all the visible custom fields.
+// When used for notes text, it should not contain the invisibleCustomFields.
+export const customFieldsJSONString = (
+  values,
+  forNoteText = false,
+  prefix = DEFAULT_CUSTOM_FIELDS_PREFIX
+) => {
+  if (Object.get(values, prefix)) {
     const clonedValues = _cloneDeep(values)
-    if (values.formCustomFields.invisibleCustomFields) {
-      clonedValues.formCustomFields.invisibleCustomFields.forEach(f =>
+    const customFieldsValues = Object.get(clonedValues, prefix)
+    if (customFieldsValues.invisibleCustomFields) {
+      customFieldsValues.invisibleCustomFields.forEach(f =>
```

Comment:
What is the type of `customFieldsValues.invisibleCustomFields`?

Reply:
array of strings

Reply:
Are you sure?
```
    objShape.invisibleCustomFields = yup.mixed().nullable().default(null)
```